### ISSUE TITLE
pubsub: make Harness.MakeSubscription no longer take a Topic arg

### ISSUE THREAD

**Issue by ijt** [google/go-cloud#861]
I'm thinking we can consolidate the interface to have a single method like this:

```go
type Harness interface {
    MakeTopicAndSubscriptions(ctx context.Context, topicName string, subNames []string) (driver.Topic, []driver.Subscription, error)
    ...
}
```
Then any connecting of the topic and subscriptions can be done within this method.

**Comment by ijt** [comment]
@vangent, what do you think?

**Comment by vangent** [comment]
We discussed this in person. I think it might work. It's a bit hard to tell without knowing what tests need.

- All else being equal, it seems better to have the tests use similar constructors as users will use. IIUC, the constructors are using names, so names should be sufficient to make subscriptions to topics (i.e., I don't think any driver should need a `driver.Topic` object in order to create a subscription to that topic).

- It's still unclear to me whether `pubsub` for record/reply is going to expect that topics/subscriptions be pre-previsioned (like `blob`, which expects the bucket to be created manually beforehand), or going to create them itself (which it can't do via the `pubsub` API; this is what `runtimevar` does). If it's going to provision things like `runtimevar`, I think it'll be cleaner for tests to separate those operations (provisioning vs creating a `pubsub` object) since you might want to do the former once for a table-driven test, and the latter once per row.

**Comment by ijt** [comment]
Okay, how about something more like this:

```go
type Harness interface {
    MakeTopicAndSubscriptionsOnServer(ctx context.Context, topicName string, subNames []string) error
    MakeLocalTopic(topicName string) driver.Topic
    MakeLocalSubscription(subName string) driver.Subscription
    Close()
}
```

**Comment by vangent** [comment]
"Server" and "Local" is misleading, because it's not always true (e.g., for `mempubsub` they are all local).

In `runtimevar` I called them `CreateXXX` for the provisioning ones and "MakeXXX" for the creating the driver object. I'm open to name bikeshedding but there's value in consistency so we whatever we pick we should update `runtimevar` to match. (`blob` doesn't do any provisioning so it just has "MakeXXX").

**Comment by jba** [comment]
I think pubsub can get away with pre-provisioning, as long as the test makes sure to drain the subscriptions when it starts to avoid flakiness due to old tests that failed in the middle.

In that case, topic and sub names would be misleading. I think we'll only ever need one topic and perhaps 2 subscriptions, so maybe
```
MakeTopic()
MakeSubscription(int) // arg can be 0 or 1
```

**Comment by jba** [comment]
Yes, you'd need a timeout. And you're right, it's not perfect, which is why in `cloud.google.com/go/pubsub` we create a new topic. But I think something like a 15s timeout would end up being less flaky than the current situation. 

But we can also just do nothing now and fix it when/if it gets annoying.

**Comment by ijt** [comment]
Also, I disagree with this approach and find it annoying to have to manually set up the testing environment even though in many cases we could easily automate it.

**Comment by jba** [comment]
I think I solved this with #1002. Although maybe we should keep it open until we have more tests.
